packages/admin-panel/src/VizBuilderApp/components/Panel.js

```diff
@@ -140,11 +140,21 @@ export const Panel = () => {
         )}
       </TabPanel>
       <TabPanel isSelected={tab === 1} Panel={PanelTabPanel}>
-        <JsonEditor
-          value={aggregate}
-          onChange={value => setTabValue('aggregate', value)}
-          onInvalidChange={handleInvalidChange}
-        />
+        {jsonToggleEnabled ? (
+          <JsonEditor
+            value={aggregate}
+            onChange={value => setTabValue('aggregate', value)}
+            onInvalidChange={handleInvalidChange}
+          />
+        ) : (
+          <AggregationDataLibrary
+            aggregate={aggregate}
+            onAggregatchange={value => {
+              setTabValue('aggregate', value);
+              setFetchJsonEditorKey(fetchJsonEditorKey + 2);
```

Comment:
Ah this just invalidates the fetch tab JsonEditor. To have this work properly you would need to add a separate variable and a `key` property to the aggregate tab JsonEditor, on line 144. 

Background: for some reason JsonEditor doesn't seem to reflect changes when you change its input props, so above I had to just invalidate the entire JsonEditor component using the key property when the value changes to get react to drop and recreate and reinitialise the JsonEditor component entirely. <-- this might be a good comment above on line 137, and maybe a `// see above` for this line.

Reply:
Emm no wonder it doesn't reflect changes when I change the input. However, after I add the json toggle, everythings work fine. The reason is that while we toggle back and forward, JsonEditor will rerender itself, which will reflect the change. I did try to remove `setFetchJsonEditorKey` on `fetch` tab and it works perfectly.
  
So I think it is fine to do so?

Reply:
Perfect, much simpler that way!